crates/weaver_health/src/attribute_advice.rs

```diff
@@ -0,0 +1,106 @@
+// Pluggable advisors
+
+use convert_case::{Boundary, Case, Casing};
+use serde::Serialize;
+use serde_json::Value;
+use weaver_resolved_schema::attribute::Attribute;
+
+use crate::{attribute_health::AttributeHealthChecker, sample::SampleAttribute};
+
+/// The severity of the advice
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub enum Severity {
+    /// An error
+    Error,
+    /// A warning
+    Warning,
+    /// Information
+    Info,
+}
```

Comment:
Is there a good reason not to use the `miette` diagnostic mechanism already in place? Miette already supports the concept of `SeverityLevel` and is already integrated with our `--future` flag.

Reply:
Oh, maybe `Severity` is not the right term here. I want to express whether the `Advice` given is a failure: like using a deprecated attribute, or a hint: like suggesting a similar attribute to use instead.

The output of the analysis is not a simple diagnostic list, it's more structured, like the diff output.

Reply:
I've swapped this out for `Advisory`: `Violation`, `Improvement`, `Information` to better express what the advisory level of the advice.